src/binary/ibc-setup/commands/init.ts

```diff
@@ -56,26 +59,55 @@ export async function run(options: Options) {
 
   const registryFilePath = path.join(options.home, registryFile);
   if (!fs.existsSync(registryFilePath)) {
-    try {
-      const registryFromRemote = await axios.get(
-        'https://raw.githubusercontent.com/confio/ts-relayer/main/demo/registry.yaml'
+    if (options.registryFrom) {
+      const existingRegistryFilePath = path.join(
+        options.registryFrom,
+        registryFile
       );
-      fs.writeFileSync(registryFilePath, registryFromRemote.data);
-    } catch (error) {
-      throw new Error(`Cannot fetch ${registryFile} from remote. ${error}`);
+      try {
+        fs.copyFileSync(existingRegistryFilePath, registryFilePath);
+        console.log(
+          `Copied existing registry from ${existingRegistryFilePath} to ${registryFilePath}.`
+        );
+      } catch (error) {
+        if (error.code === 'ENOENT') {
+          throw new Error(
+            `No such file: ${existingRegistryFilePath}. Make sure that "--registry-from" points at existing relayer's home dir.`
+          );
+        } else {
+          throw error;
+        }
+      }
+    } else {
+      try {
+        const registryFromRemote = await axios.get(
+          'https://raw.githubusercontent.com/confio/ts-relayer/main/demo/registry.yaml'
+        );
+        fs.writeFileSync(registryFilePath, registryFromRemote.data);
+        console.log(`Pulled default ${registryFile} from remote.`);
+      } catch (error) {
+        throw new Error(`Cannot fetch ${registryFile} from remote. ${error}`);
+      }
     }
   } else if (!fs.lstatSync(registryFilePath).isFile()) {
     throw new Error(`${registryFilePath} must be a file.`);
   }
 
+  if (!options.src || !options.dest) {
+    console.log(
+      `Exited earlier. To generate ${appFile}, run the command again with "src" and "dest" chains defined.`
+    );
+    return;
+  }
```

Comment:
In the beginning, I added and additional `--download-only` option, however:
- it introduced a few additional if statements and early returns what made the code more complex
- I find it hard to explain it nicely in the spec

Instead, I thought it would be better to remove `--download-only` flag to make the command more resilient. I changed the `src` and `dest` options to be optional so now `ibc-setup init` is equivalent of init with `--download-only` flag.

What do you think? Please check out the updated spec: https://github.com/confio/ts-relayer/pull/166/files#diff-04d092475af8d9a0591e7c3991d8789f2a9875595614041194d6960f7c816b17

Reply:
Also, it solves the issue we recently spoke about that the `src` and `dest` are required and the user needs to know chain names upfront.

Reply:
Hmmm... so if I want to use a chain not in the default registry, I need to:

* `ibc-setup init`
* `vi ~/.ibc-setup/registry.yaml`
* `ibc-setup init --src my-custom-chain --dest cosmos-hub`

Is that right?

We should definitely document it if so.

Reply:
I would change the log message.

> Exited earlier. To generate ${appFile}, run the command again with "src" and "dest" chains defined

Better, something to explain the process:

"Registry file downloaded to ${home}/registry.yaml. Please edit that file and add any chains you wish. Then complete the initialization by running `ibc-setup init --src <chain-1> --dest <chain-2>`"

Reply:
1aa47bf

>We should definitely document it if so.

Do you think the changes in the spec are enough?
https://github.com/confio/ts-relayer/pull/166/files#diff-04d092475af8d9a0591e7c3991d8789f2a9875595614041194d6960f7c816b17

Reply:
Those are good comments.

It would be nice to have some docs in README. eg. after "quick start" a brief flow for connecting this to your custom chain. Showing the flow - `init`, edit file, `init --src --dest`, fund relater, `ics20`, start relayer...